### ISSUE TITLE
/list_venues command is not working

### ISSUE THREAD

**Issue by tejaswankalluri** [abinpaul1/BookMyShow-ticket-notifier-telegram-bot#4]
### When i am using list venues /list_venues HYD. server is crashing

- Error happening because of no error handling on ` api_get_venues_for_region ()` function
- I Have checked the get venues API which is returning empty data

https://github.com/abinpaul1/BookMyShow-ticket-notifier-telegram-bot/blob/7758ade81135e0082e8ded24415c13fd64c05efe/src/bms_helper.rs#L93

![image](https://user-images.githubusercontent.com/49343044/200844676-daf63bf9-c734-4eb0-bdc9-f6d3ae695c55.png)

**Comment by abinpaul1** [comment]
Yeah, the endpoint isn't working anymore. I have a fix to check for empty in my local as a workaround. (Will push this in a couple days when I get time ) Ideally we should find a working endpoint again.

**Comment by tejaswankalluri** [comment]
Ha, I dont know rust being a nodejs dev, I just commented the API request in that function. that's the temporary solution I got, btw rust is pretty hard to understand 😢.

**Comment by abinpaul1** [comment]
Yeah, rust is tricky. I wouldn't start with looking at this codebase (this was my learn rust project 😅 ).

I've pushed the change to check for empty. I'll keep the issue open till I get time to find the new endpoint

**Comment by tejaswankalluri** [comment]
No Logging is working after changes in the code's last commit
## the code is running but no logs

![image](https://user-images.githubusercontent.com/49343044/203244155-962d0756-d6de-4c86-bc3e-272ff596880e.png)

**Comment by Yooosless** [comment]
I have resolved logs issue locally (now i can see logs using new tracing crate)

coming to list venues issue, Bookmyshow is shielded by cloudflare now and we cannot hit it an endpoint:)
```
2026-06-27T17:39:57.271129Z  INFO book_my_show_notifier_bot::notification_controller: list_venues_for_location: location_code: HYD
2026-06-27T17:39:57.271249Z  INFO book_my_show_notifier_bot::bms_helper: BmsHelper::api_get_venues_for_region: HYD
2026-06-27T17:39:57.723032Z  INFO book_my_show_notifier_bot::bms_helper: RAW BMS RESPONSE: {
  "description": "blocked through send",
  "errcode": "0xa",
  "legacy": null,
  "message": "Sorry, we are facing some unexpected issues at our end. Please try again after sometime - 0xa",
  "statusCode": 400
}
```

**Comment by jasv11** [comment]
> I have resolved logs issue locally (now i can see logs using new tracing crate)
> 
> coming to list venues issue, Bookmyshow is shielded by cloudflare now and we cannot hit it an endpoint:)
> 
> ```
> 2026-06-27T17:39:57.271129Z  INFO book_my_show_notifier_bot::notification_controller: list_venues_for_location: location_code: HYD
> 2026-06-27T17:39:57.271249Z  INFO book_my_show_notifier_bot::bms_helper: BmsHelper::api_get_venues_for_region: HYD
> 2026-06-27T17:39:57.723032Z  INFO book_my_show_notifier_bot::bms_helper: RAW BMS RESPONSE: {
>   "description": "blocked through send",
>   "errcode": "0xa",
>   "legacy": null,
>   "message": "Sorry, we are facing some unexpected issues at our end. Please try again after sometime - 0xa",
>   "statusCode": 400
> }
> ```

Just to confirm, is the app no longer working as intended?

**Comment by Yooosless** [comment]
> > I have resolved logs issue locally (now i can see logs using new tracing crate)
> > 
> > coming to list venues issue, Bookmyshow is shielded by cloudflare now and we cannot hit it an endpoint:)
> > 
> > ```
> > 2026-06-27T17:39:57.271129Z  INFO book_my_show_notifier_bot::notification_controller: list_venues_for_location: location_code: HYD
> > 2026-06-27T17:39:57.271249Z  INFO book_my_show_notifier_bot::bms_helper: BmsHelper::api_get_venues_for_region: HYD
> > 2026-06-27T17:39:57.723032Z  INFO book_my_show_notifier_bot::bms_helper: RAW BMS RESPONSE: {
> >   "description": "blocked through send",
> >   "errcode": "0xa",
> >   "legacy": null,
> >   "message": "Sorry, we are facing some unexpected issues at our end. Please try again after sometime - 0xa",
> >   "statusCode": 400
> > }
> > ```
> 
> Just to confirm, is the app no longer working as intended?

nope, now everything is behind cloudflare and this is not built in a way to overcome that.

**Comment by jasv11** [comment]
> > > I have resolved logs issue locally (now i can see logs using new tracing crate)
> > > coming to list venues issue, Bookmyshow is shielded by cloudflare now and we cannot hit it an endpoint:)
> > > ```
> > > 2026-06-27T17:39:57.271129Z  INFO book_my_show_notifier_bot::notification_controller: list_venues_for_location: location_code: HYD
> > > 2026-06-27T17:39:57.271249Z  INFO book_my_show_notifier_bot::bms_helper: BmsHelper::api_get_venues_for_region: HYD
> > > 2026-06-27T17:39:57.723032Z  INFO book_my_show_notifier_bot::bms_helper: RAW BMS RESPONSE: {
> > >   "description": "blocked through send",
> > >   "errcode": "0xa",
> > >   "legacy": null,
> > >   "message": "Sorry, we are facing some unexpected issues at our end. Please try again after sometime - 0xa",
> > >   "statusCode": 400
> > > }
> > > ```
> > 
> > 
> > Just to confirm, is the app no longer working as intended?
> 
> nope, now everything is behind cloudflare and this is not built in a way to overcome that.

Thanks for the quick reply, Are there any alternative apps or projects you'd recommend that provide similar functionality?
